src/SAML2/XML/AbstractConvertable.php

```diff
@@ -27,6 +29,36 @@ public function __toString(): string
     }
 
 
+    /**
+     * Serialize this XML chunk
+     *
+     * @return string The serialized chunk.
+     */
+    public function serialize(): string
+    {
+        return serialize($this->toXML()->ownerDocument->saveXML());
+    }
+
+
+    /**
+     * Un-serialize this XML chunk.
+     *
+     * @param string $serialized The serialized chunk.
+     * @return void
+     *
+     * Type hint not possible due to upstream method signature
+     */
+    public function unserialize($serialized): void
+    {
+        $doc = DOMDocumentFactory::fromString(unserialize($serialized));
```

Comment:
... and the same here. No need to call `unserialize()`.

Reply:
And done